### PR TITLE
Generate project status at end of process

### PR DESCRIPTION
**Pull request by AJenbo**

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+make -f MakefileVC
+objdump -a -d Diablo.exe | cut -f 3- | sed -E s/0x[0-9a-f]+/0xaddress/g > /tmp/Diablo.asm
```
Comment (seritools) [inline build.sh]: That `sed` command will remove all hexadecimal numbers, which also includes stack offsets.
I'm not sure if this won't remove too much info out of the asm file.

**Comment by seritools** [comment]
The way it is right now, this filters out all constants, stack offsets, etc.

I've been thinking about this a bit, and the only way to make this as conservative as possible would be to find a way to strip out only addresses and nothing else. However that's easily said than done. Does someone have an idea for that?

**Comment by seritools** [comment]
Okay, after puzzling with @AJenbo we've got the following:

objdump calculates the resulting address for relative jumps, and there doesn't seem to be a way to prevent that easily.

- Grab only the contents of the `.text` segment (a.k.a. the code) and feed only that to objdump. Hopefully that way it doesn't know its position and displays the relative jump as just that - a relative jump. These should be exactly the same in devilution
- All absolute addresses are marked with a `*` before them, so these should be the only ones to be removed
- You'd probably read the right size of the `.text` segment in devilution to grab the right portions of the executable. For the original binary it's (0x400..0x77A00). objdump should be able to output this info via the `-h` switch.

This is not tested yet, but could work since it makes sure only absolute addresses are removed.

**Comment by seritools** [comment]
I've just pushed devilution-comparer 0.3 which has a new mode designed for riivaja, so running
`devilution-comparer generate-full Diablo_orig.exe --orig-file --no-mem-disp` and putting the output file of that into the image, then running `devilution-comparer generate-full Diablo.exe --no-mem-disp` on each CI run and diffing both files should work fine now :)

Oh yeah diff has a `--suppress-common-lines`, which makes the diff file output smaller by the way (since we're just counting the lines starting with + and - anyways).

The remaining question is whether to build devilution-comparer within the container or having the image creator/user supply the linux version. (I'd say supply it to keep the container size small)

**Comment by AJenbo** [comment]
Updated PR, --suppress-common-lines is not supported by the version of diff being used.

File: README.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ Docker build system for devilution using the original tools.
 
 In order to verify the accuracy of the source code and to generate a binary that is as close as possible to the original one, it is necessary to compile it using the original toolchain. This tool generates a docker image that can be used to compile devilution on any platform with that toolchain.
 
-This build system is used in the devilution CI system as well in order to make sure that all changes can be compiled with the original toolchain.
+This build system is used in the devilution CI system as well in order to make sure that all changes can be compiled with the original toolchain and also calculate the accuracy of the binery.
```
Comment (seritools) [inline README.md]: *binary ;)

File: README.md [hunk 3]
```diff
@@ -22,7 +22,7 @@ docker run -v $(pwd):/root/devilution -e MAKE_BUILD=pdb diasurgical/riivaaja
 
 ## Building a new image
 
-Place the files for Visual C++ 5 SP3 (DevStudio_5.10) and Visual C++ SP5 PP5 (VS6) in the Riivaaja root folder. Make sure the versions of VC actually support optimizations (Professional and higher).
+Place the files for Visual C++ 5 SP3 (DevStudio_5.10) and Visual C++ SP5 PP5 (VS6) in the Riivaaja root folder. Make sure the versions of VC actually support optimizations (Professional and higher). Use Devilution-comparer to generate a orig_full.asm and put it in the root.
```
Comment (seritools) [inline README.md]: Could you add that this is optional and only needed if you want to get the comparation results?

**Comment by seritools** [comment]
🎉

**Pull request merged**